src/components/frame/v5/pages/StreamingPaymentsPage/partials/StreamingActionsTable/StreamingActionsTable.tsx

```diff
@@ -0,0 +1,175 @@
+import { ArrowCircleDown, FilePlus, ShareNetwork } from '@phosphor-icons/react';
+import {
+  type SortingState,
+  type Row,
+  getPaginationRowModel,
+} from '@tanstack/react-table';
+import clsx from 'clsx';
+import React, { useState, type FC } from 'react';
+import { defineMessages } from 'react-intl';
+import { generatePath, useNavigate } from 'react-router-dom';
+
+import MeatballMenuCopyItem from '~common/ColonyActionsTable/partials/MeatballMenuCopyItem/MeatballMenuCopyItem.tsx';
+import { APP_URL } from '~constants';
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { useMobile, useTablet } from '~hooks';
+import {
+  COLONY_ACTIVITY_ROUTE,
+  COLONY_HOME_ROUTE,
+  TX_SEARCH_PARAM,
+} from '~routes';
+import { formatText } from '~utils/intl.ts';
+import { MEATBALL_MENU_COLUMN_ID } from '~v5/common/Table/consts.ts';
+import { Table } from '~v5/common/Table/Table.tsx';
+
+import {
+  type StreamingTableFieldModel,
+  type StreamingActionTableFieldModel,
+} from '../StreamingPaymentsTable/types.ts';
+
+import { useStreamingActionsTableColumns } from './hooks.tsx';
+import useRenderSubComponent from './partials/StreamingActionMobileItem/hooks.tsx';
+import useRenderRowLink from './useRenderRowLink.tsx';
+import { orderActions } from './utils.ts';
+
+const displayName =
+  'pages.StreamingPaymentsPage.partials.StreamingActionsTable.StreamingActionsTable';
+
+const MSG = defineMessages({
+  loadMoreStreams: {
+    id: `${displayName}.loadMoreStreams`,
+    defaultMessage: 'Load more streams',
+  },
+});
+
+interface StreamingActionsTableProps {
+  actionRow: Row<StreamingTableFieldModel>;
+  isLoading: boolean;
+}
+
+const StreamingActionsTable: FC<StreamingActionsTableProps> = ({
+  actionRow,
+  isLoading,
+}) => {
+  const {
+    colony: { name: colonyName },
+  } = useColonyContext();
+  const { original } = actionRow;
+  const isTablet = useTablet();
+  const isMobile = useMobile();
+  const columns = useStreamingActionsTableColumns();
+  const [sorting, setSorting] = useState<SortingState>([
+    {
+      id: 'totalStreamedAmount',
+      desc: true,
+    },
+  ]);
+  const navigate = useNavigate();
+  const getMenuProps = ({ original: { transactionId } }) => ({
+    items: [
+      {
+        key: '1',
+        label: formatText({ id: 'activityFeedTable.menu.view' }),
+        icon: FilePlus,
+        onClick: () => {
+          navigate(
+            `${window.location.pathname}?${TX_SEARCH_PARAM}=${transactionId}`,
+            {
+              replace: true,
+            },
+          );
+        },
+      },
+      {
+        key: '2',
+        label: formatText({ id: 'activityFeedTable.menu.share' }),
+        renderItemWrapper: (itemWrapperProps, children) => (
+          <MeatballMenuCopyItem
+            textToCopy={`${APP_URL.origin}/${generatePath(COLONY_HOME_ROUTE, {
+              colonyName,
+            })}${COLONY_ACTIVITY_ROUTE}?${TX_SEARCH_PARAM}=${transactionId}`}
```

Comment:
Could we create a helper function in the `utils` to retrieve this url with search parameters? You could make use of `setQueryParamOnUrl` in this function if parameters are given, as an idea

This part seems to be duplicated across the app and I think it makes sense to clean it a bit up 😄 but can be done also at a later point